package tests;

import base.BaseTest;
import io.restassured.RestAssured;
import io.restassured.response.Response;
import org.testng.Assert;
import org.testng.annotations.Test;

public class AuthAPITest extends BaseTest {

    @Test(dataProvider = "registerData")
    public void testRegister(String testType, String testName, String baseUrl, String endpoint,
                             String username, String password) throws Exception {

        System.out.println("Running Register test -> " + testName);

        Response response = RestAssured.given()
                .baseUri(baseUrl)
                .contentType("application/json")
                .body("{\"username\":\"" + username + "\", \"password\":\"" + password + "\"}")
                .when()
                .post(endpoint)
                .then()
                .extract()
                .response();

        System.out.println("Response: " + response.asString());

        String actualResult = response.asString().contains("success") ? "PASS" : "FAIL";

        int rowIndex = excel.findRowIndexByTestName("Tests", testName);
        if (rowIndex != -1) {
            excel.setCellData("Tests", rowIndex, 6, actualResult); // Column G = Result
        } else {
            System.out.println("TestName not found in Excel: " + testName);
        }

        Assert.assertEquals(actualResult, "PASS");
    }
}
